### ISSUE TITLE
Add eval scenario: R3 Knowledge Duplication via duplicated validation logic

### ISSUE THREAD

**Issue by hyhmrright** [hyhmrright/brooks-lint#4]
## Scenario

**Mode:** PR Review (Mode 1)  
**Risk:** R3 Knowledge Duplication

## Input

A PR that adds a third copy of the same email validation regex in a different service, when two identical copies already exist in the codebase.

```python
# users/validators.py  (copy 1)
EMAIL_RE = re.compile(r'^[\w.+-]+@[\w-]+\.[a-z]{2,}$')

# orders/validators.py  (copy 2 — already exists)
EMAIL_RE = re.compile(r'^[\w.+-]+@[\w-]+\.[a-z]{2,}$')

# notifications/validators.py  (copy 3 — added in this PR)
EMAIL_RE = re.compile(r'^[\w.+-]+@[\w-]+\.[a-z]{2,}$')
```

## Expected finding

Brooks-lint should flag this as 🔴 R3 Knowledge Duplication:
- Symptom: same decision (email format rule) expressed in three separate files
- Source: The Pragmatic Programmer — DRY principle
- Consequence: one copy gets updated, the others drift silently
- Remedy: extract to a shared `common/validators.py`

## How to contribute

1. Add this scenario to `evals/evals.json` as the next sequential entry
2. See `CONTRIBUTING.md` for the exact JSON format
3. Open a PR — no other changes needed

**Comment by hyhmrright** [comment]
This scenario is already covered by the `r3-critical-go` eval in `evals/evals.json`: email validation logic duplicated across three files (`user.go`/`admin.go`/`auth.go`), expected to flag 🔴 R3 Knowledge Duplication citing Pragmatic Programmer (DRY) + Fowler (Duplicate Code). Same shape as the Python example here, so the coverage gap this issue targeted is closed. Thanks!
